sklearn/cross_decomposition/_pls.py

```diff
@@ -530,7 +530,7 @@ class PLSRegression(_PLS):
     ----------
     n_components : int, default=2
         Number of components to keep. Should be in `[1, min(n_samples,
-        n_features, n_targets)]`.
+        n_features)]`.
```

Comment:
Looking at the code it is even only `[1, n_features]` for `PLSRegression

Reply:
> Looking at the code it is even only `[1, n_features]` for `PLSRegression

I read the code of function "plsregress" in MATLAB and do some experiments of PLSRegression in sklearn, now I am very sure the n_components should be in [1, min(n_samples - 1, n_features)] !

Reply:
Reading the code in MATLAB is not relevant here. Here is the part in the scikit-learn codebase:

https://github.com/scikit-learn/scikit-learn/blob/10dbc142bd17ccf7bd38eec2ac04b52ce0d1009e/sklearn/cross_decomposition/_pls.py#L241-L254

In the case of regression, the maximum rank is set to `X.shape[1]` and we raise if the number of components is strictly greater.

@li1127217ye Could you provide the code snippet where we would do `min(n_samples - 1, n_features)` in case that I missed it?